### ISSUE TITLE
Lighthouse improvements?

### ISSUE THREAD

**Issue by MichaelNussbaumerGOWEST** [TYPO3-Headless/nuxt-typo3#243]
I currently have a huge nuxt-typo3 application where I have pretty bad Pagespeed scores. I would say the score doesn't reflect the real life performance, but as Google considers the score in some way this is basically not an good argument. I think one of the main issues is that the main JS got pretty big, I am curious if someone also has a bit project or any ideas how to start to optimize this.

In smaller projects we don't have a big problem with Lighthouse. 

One of my ideas was to change the approach of importing components and lazy load them only if they are really in use on the current page. I don't know if this is even possible, but if, this would maybe also a good thing for the core application I think..

What downsides am i missing maybe?

**Comment by mercs600** [comment]
Yes, nuxt-typo3 is relatively cheap if it comes to kB cost, but if you have a lot of custom components/plugins then it could be a problem. There are few options you can do:
1. Disable all nuxt-typo3 components and add only which you need https://github.com/TYPO3-Headless/nuxt-typo3/blob/master/lib/options.js#L14
2. You can add components asynchronously, but do it wisely, because a lot of single small request may not be preferable 
```js
const CeMycustomElement = () =>
  import('~/components/CeMycustomElement' /* webpackChunkName: "CeMycustomElement" */)
```
3. You don't have to define all components as global (for example in `plugins/components.js` file) - depends on project ofc, but you can define components only for specific backend layout or for specific page - not all pages need to be dynamic, if you know the route name of page you can create specific static page file and define components there 
4. I highly recommend the use vue-lazy-hydration https://github.com/maoberlehner/vue-lazy-hydration

**Comment by MichaelNussbaumerGOWEST** [comment]
Thanks for the feedback. I also meant 2. with component lazy loading. I will give it a try after we tested if Bing is also able to handle the JS generated content correctly, same as google does it. 

Because our customer have 7% organic traffic from bing, we can't ignore it in this case.

Sadly we have around 70 custom components and 5 backend layouts. I think we are not able to isolate which component is used on which backend layout. 

The page also has 17.000 pages, so I think the only way to go is to load the components asynchronously. I think it is way better than the huge app.js right now.

Do you have any experience about asynchron loading of components and bing indexing? If not, I will update this issue after my test got indexed and what will be the outcome.

**Comment by mercs600** [comment]
There is no difference If you use nuxt universal mode (ssr) -  when you load components asynchronously then you should get components content in the source of your page 😄 We use  asynchronously loading a lot.

**Comment by MichaelNussbaumerGOWEST** [comment]
FYI: I implemented vue-lazy-hydration and component lazy loading, also I analyzed my build and got rid of some libaries. I was able to get decent results for mobile now considering the size of the site.

Additionally we analyzed the use of all components and made a list which one we can might drop as they are barely used.

My suggestions would be to add maybe a small documentation with some hints how to make improvements in this case.

Also, is there a discord server or something for nuxt-typo3, where we can maybe share some knowledge in the commuinity? I am on some discord servers for other projects, and it is pretty helpful to have a place like that to communicate with other developers.
